### PR TITLE
feat: one-click graphQL query & mutation generation

### LINKED ISSUE #4371: [feature]: One-click GraphQL Query & Mutation Generation
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

There is a lack of functionality for automatically generating query and mutation templates for selected data types or operations from the documentation (as it is done, for example, in Altair GraphQL).

### Why should this be worked on?

It's not always convenient to create a schema from scratch, even with autocomplete. It would be easier if we could open the documentation and insert an entire schema (query/mutation/fragment), with the option to remove unnecessary parts. Given that schemas can be very complex with deep nesting, a useful feature would be a setting to control the 'max depth' level of this nesting when adding a schema.

### PR DESCRIPTION
**Pull request by bogking**
Closes #4371

### PR Description
This PR adds functionality to be able to automatically generate query and mutation templates for selected data types or operations from the documentation and adds a setting to control the 'max depth' level of this nesting when adding a schema.

### What's changed
- Updated Query.vue to be able to display the query generated from the documentation.  
- Updated Sidebar.vue to be able to include a tab for getting queries from the documentation and a button to do so.  
- Updated settings.ts to contain a setting to set the maximum nesting depth for the queries generated from the documentation.  
- Updated DispatchingStore.ts slightly to enable dispatches to be able to work with the rest of the code.  
- Updated index.ts to be able to integrate the graphql schema templates and requests with the changes made to be able to get queries from the documentation.   
- Created queryBuilder.ts as a helper file to be able to generate a graphql query based on the documentation. Contains functions to generate and build the query.  
- Created har.ts to be able to parse the GQL request from the HoppGQLRequest and convert it to a HTTP Archive or HAR request.  
- Created index.ts to be able to use the HTTPSnippet library to make code snippets for the languages listed in codegenDefinitions from the HAR requests made through the har.ts file.

### CODE REVIEW

**Comment by Olly4k** [comment]
The implementation looks good. However, I just want to check whether you've eliminated all of your local hosting changes. There seems to be some left over changes in the docker files that you should probably clean up.

File: docker-compose.deploy.yml [hunk 1]
```diff
@@ -6,9 +6,9 @@ services:
     image: postgres:15
     user: postgres
     environment:
-      POSTGRES_USER: postgres
       POSTGRES_PASSWORD: testpass
       POSTGRES_DB: hoppscotch
+      POSTGRES_DB: test
```
Comment (AndrewBastin) [inline docker-compose.deploy.yml:11]: Revert this change.

File: packages/hoppscotch-data/src/graphql/index.ts
Comment (AndrewBastin) [inline packages/hoppscotch-data/src/graphql/index.ts:1]: Revert this file back, the contents given here seems to be copied over from `hoppscotch-common`. `hoppscotch-data` changes should be backwards compatible (which this is not currently).

**Review by AndrewBastin — CHANGES_REQUESTED** [review]

**Comment by anwarulislam** [comment]
This PR is incomplete and has invalid implementation. Following the issue in a different PR for a more accurate and complete solution. [#4626](https://github.com/hoppscotch/hoppscotch/pull/4626)

**Pull request closed without merging**